draft-ietf-oauth-v2-1.md

```diff
@@ -958,35 +960,37 @@ Extension grant types MAY define additional endpoints as needed.
 ## Authorization Endpoint
 
 The authorization endpoint is used to interact with the resource
-owner and obtain an authorization grant.  The authorization server
-MUST first verify the identity of the resource owner.  The way in
-which the authorization server authenticates the resource owner
+owner and obtain an authorization grant.  The AS
+MUST first authenticate the resource owner.  The way in
+which the AS authenticates the resource owner
 (e.g., username and password login, session cookies) is beyond the
 scope of this specification.
 
-The means through which the client obtains the location of the
+The means through which the client obtains the URL of the
 authorization endpoint are beyond the scope of this specification,
-but the location is typically provided in the service documentation,
+but the URL is typically provided in the service documentation,
 or in the authorization server's metadata document ({{RFC8414}}).
 
-The endpoint URI MAY include an "application/x-www-form-urlencoded"
+The authorization endpoint URL:
+
+- MUST NOT include a fragment component;
+- MAY include an "application/x-www-form-urlencoded"
 formatted (per Appendix B) query component ({{RFC3986}} Section 3.4),
-which MUST be retained when adding additional query parameters.  The
-endpoint URI MUST NOT include a fragment component.
+which MUST be retained when adding additional query parameters.
 
-The authorization server MUST support the use of the HTTP `GET`
-method {{RFC7231}} for the authorization endpoint and MAY support the
-use of the `POST` method as well.
+The authorization endpoint
+MUST support GET requests
+and MAY support POST requests.
 
-The authorization server MUST ignore unrecognized request parameters. 
+The authorization endpoint MUST ignore unrecognized request parameters.
 
 Request and response parameters
 defined by this specification MUST NOT be included more than once. 
 Parameters sent without a value MUST be treated as if they were
 omitted from the request.
 
-An authorization server that redirects a request potentially containing 
-user credentials MUST avoid forwarding these user credentials accidentally 
+An authorization endpoints that redirects a request potentially containing
+user credentials MUST NOT forward these credentials to untrusted parties
```

Comment:
I don't think adding "untrusted parties" here is beneficial

Reply:
Can we say that "it MUST NOT forward these credentials" never, in any case?

Reply:
That gets a bit tricky, because it might need to "forward" the credentials to itself, and then what are the boundaries of its "self" becomes challenging to describe concisely. I think that's the reason the original text said "accidentally".

Reply:
@aaronpk "MUST avoid to accidentally" sounds like [SHOULD CONSIDER](https://datatracker.ietf.org/doc/html/rfc6919#section-2). IMHO does not  provide any interoperability guidance because it is related to an unintended behavior. 

In general, I'd track down RFC6919-sounding statements in this document and address them before AD / IESG review will ask us to do it.

> I don't think adding "untrusted parties" here is beneficial

imho  helps avoiding the fallacies of "accidentally", since it suggests the AS to internally define a trusted boundary for credentials.